plugins/qlcplus_4.12.1/import.js

```diff
@@ -0,0 +1,913 @@
+    const cap = {
+      type: `Effect`,
+      effectName: `` // set it first here so effectName is before speedStart/speedEnd
+    };
+    cap.effectName = importHelpers.getSpeedGuessedComment(capabilityName, cap);
+
+    if (/\bsound\b/i.test(cap.effectName)) {
+      cap.soundControlled = true;
+    }
+
+    return cap;
+  },
+  Maintenance: ({ capabilityName }) => ({
+    type: `Maintenance`,
+    comment: capabilityName
+  }),
+  Pan: ({ channelName, capabilityName, panMax }) => {
+    if (channelName.match(/continuous/i)) {
+      const cap = {
+        type: `PanContinuous`
+      };
+      cap.comment = importHelpers.getSpeedGuessedComment(capabilityName, cap);
+      return cap;
+    }
+
+    return Object.assign(importHelpers.getPanTiltCap(`Pan`, panMax), {
+      comment: capabilityName
+    });
+  },
+  Tilt: ({ channelName, capabilityName, tiltMax }) => {
+    if (channelName.match(/continuous/i)) {
+      const cap = {
+        type: `TiltContinuous`
+      };
+      cap.comment = importHelpers.getSpeedGuessedComment(capabilityName, cap);
+      return cap;
+    }
+
+    return Object.assign(importHelpers.getPanTiltCap(`Tilt`, tiltMax), {
+      comment: capabilityName
+    });
+  },
+  Prism: ({ capabilityName }) => ({
+    type: `Prism`,
+    comment: capabilityName
+  }),
+  Shutter: ({ capabilityName }) => {
+    const cap = {
+      type: `ShutterStrobe`
+    };
+
+    if (capabilityName.match(/^(?:Blackout|(?:Shutter |Strobe )?Closed?)$/i)) {
+      cap.shutterEffect = `Closed`;
+      return cap;
+    }
+
+    if (capabilityName.match(/^(?:(?:Shutter |Strobe )?Open|Full?)$/i)) {
+      cap.shutterEffect = `Open`;
+      return cap;
+    }
+
+    if (capabilityName.match(/puls/i)) {
+      cap.shutterEffect = `Pulse`;
+    }
+    else if (capabilityName.match(/ramp\s*up/i)) {
+      cap.shutterEffect = `RampUp`;
+    }
+    else if (capabilityName.match(/ramp\s*down/i)) {
+      cap.shutterEffect = `RampDown`;
+    }
+    else {
+      cap.shutterEffect = `Strobe`;
+    }
+
+    if (capabilityName.match(/random/i)) {
+      cap.randomTiming = true;
+    }
+
+    cap.comment = importHelpers.getSpeedGuessedComment(capabilityName, cap);
+
+    return cap;
+  },
+  Speed: ({ channelName, capabilityName }) => {
+    const cap = {};
+
+    if (channelName.match(/pan(?:\/)?tilt/i)) {
+      cap.type = `PanTiltSpeed`;
+    }
+    else {
+      cap.type = `Speed`;
+    }
+
+    cap.comment = importHelpers.getSpeedGuessedComment(capabilityName, cap);
+
+    return cap;
+  }
+};
+
+/**
+ * Adds a QLC+ channel to the OFL fixture's availableChannels object.
+ * @param {object} fixture The OFL fixture object.
+ * @param {object} qlcPlusChannel The QLC+ channel object.
+ * @param {object} qlcPlusFixture The QLC+ fixture object.
+ */
+function addOflChannel(fixture, qlcPlusChannel, qlcPlusFixture) {
+  const channel = {
+    fineChannelAliases: [],
+    dmxValueResolution: `8bit`,
+    defaultValue: parseInt(qlcPlusChannel.$.Default) || 0
```

Comment:
Should really every channel get a default value of 0 if it isn't set from QLC+?

Reply:
Why not? The default value in QLC+ is well-defined as 0.